gcp-auth-extension/src/main/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProvider.java

```diff
@@ -79,6 +105,34 @@ public int order() {
     return Integer.MAX_VALUE - 1;
   }
 
+  private static SpanExporter customizeSpanExporter(
+      SpanExporter exporter, GoogleCredentials credentials) {
+    if (isSignalTargeted(SIGNAL_TYPE_TRACES)) {
+      return addAuthorizationHeaders(exporter, credentials);
+    }
+    return exporter;
+  }
+
+  private static MetricExporter customizeMetricExporter(
+      MetricExporter exporter, GoogleCredentials credentials) {
+    if (isSignalTargeted(SIGNAL_TYPE_METRICS)) {
+      return addAuthorizationHeaders(exporter, credentials);
+    }
+    return exporter;
+  }
+
+  // Checks if the auth extension is configured to target the passed signal for authentication.
+  private static boolean isSignalTargeted(String checkSignal) {
+    String userSpecifiedTargetedSignals =
+        ConfigurableOption.GOOGLE_OTEL_AUTH_TARGET_SIGNALS.getConfiguredValueWithFallback(
```

Comment:
Any reason why you didn't use `configProperties` to read the configuration? It is possible to add additional properties to `configProperties` like for example agent allows adding configuration in a properties file https://opentelemetry.io/docs/zero-code/java/agent/configuration/#configuration-file or the otel spring starter adds spring properties. By reading system properties and env yourself your extension won't play nice with these additional property soruces.

Reply:
That's a good idea. This was an oversight, I wasn't thinking of use-cases like declarative configs when this extension was first added.

I'll tackle this in #1903 to keep this PR's scope limited.